services/sqs/src/main/java/software/amazon/awssdk/services/sqs/internal/batchmanager/ChangeMessageVisibilityBatchManager.java

```diff
@@ -0,0 +1,156 @@
+            List<Either<IdentifiableMessage<ChangeMessageVisibilityResponse>, IdentifiableMessage<Throwable>>> mappedResponses =
+                new ArrayList<>();
+            batchResponse.successful().forEach(
+                batchResponseEntry -> {
+                    IdentifiableMessage<ChangeMessageVisibilityResponse> response = createChangeMessageVisibilityResponse(
+                        batchResponseEntry, batchResponse);
+                    mappedResponses.add(Either.left(response));
+                });
+            batchResponse.failed().forEach(batchResponseEntry -> {
+                IdentifiableMessage<Throwable> response = changeMessageVisibilityCreateThrowable(batchResponseEntry);
+                mappedResponses.add(Either.right(response));
+            });
+            return mappedResponses;
+        };
+    }
+
+    private static IdentifiableMessage<ChangeMessageVisibilityResponse> createChangeMessageVisibilityResponse(
+        ChangeMessageVisibilityBatchResultEntry successfulEntry, ChangeMessageVisibilityBatchResponse batchResponse) {
+        String key = successfulEntry.id();
+        ChangeMessageVisibilityResponse.Builder builder = ChangeMessageVisibilityResponse.builder();
+        if (batchResponse.responseMetadata() != null) {
+            builder.responseMetadata(batchResponse.responseMetadata());
+        }
+        if (batchResponse.sdkHttpResponse() != null) {
+            builder.sdkHttpResponse(batchResponse.sdkHttpResponse());
+        }
+        ChangeMessageVisibilityResponse response = builder.build();
+        return new IdentifiableMessage<>(key, response);
+    }
+
+    private static IdentifiableMessage<Throwable> changeMessageVisibilityCreateThrowable(BatchResultErrorEntry failedEntry) {
+        String key = failedEntry.id();
+        AwsErrorDetails errorDetailsBuilder = AwsErrorDetails.builder().errorCode(failedEntry.code())
+                                                             .errorMessage(failedEntry.message()).build();
+        Throwable response = SqsException.builder().awsErrorDetails(errorDetailsBuilder).build();
+        return new IdentifiableMessage<>(key, response);
+    }
+
+    private static BatchKeyMapper<ChangeMessageVisibilityRequest> changeMessageVisibilityBatchKeyMapper() {
+        return request -> request.overrideConfiguration().map(overrideConfig -> request.queueUrl() + overrideConfig.hashCode())
+                                 .orElse(request.queueUrl());
+    }
+
+    @Override
+    protected CompletableFuture<ChangeMessageVisibilityBatchResponse> batchAndSend(
+        List<IdentifiableMessage<ChangeMessageVisibilityRequest>> identifiedRequests, String batchKey) {
+        return changeMessageVisibilityBatchAsyncFunction(sqsAsyncClient).batchAndSend(identifiedRequests, batchKey);
+    }
+
+    @Override
+    protected String getBatchKey(ChangeMessageVisibilityRequest request) {
+        return changeMessageVisibilityBatchKeyMapper().getBatchKey(request);
```

Comment:
Can we just implement the logic directly in this method instead of using the lambda returned by `changeMessageVisibilityBatchKeyMapper()`? I think we can just get rid of these interfaces (`BatchAndSend`, `BatchKeyMapper`, `BatchResponseMapper`)

```java
    @Override
    protected String getBatchKey(ChangeMessageVisibilityRequest request) {
        return request.overrideConfiguration()
                      .map(overrideConfig -> request.queueUrl() + overrideConfig.hashCode())
                      .orElse(request.queueUrl());
    }
```

similar comment for `mapBatchResponse()`

Reply:
+1 , removed these classes , Thanks